agent/api/task/task.go

```diff
@@ -2340,64 +2340,119 @@ func (task *Task) dockerLinks(container *apicontainer.Container, dockerContainer
+	}
+	return ingressDockerPortMap, ingressContainerPortSet, err
+}
+
 // dockerPortMap creates a port binding map for
-// (1) Ingress listeners for the service connect AppNet container in the service connect enabled bridge network mode task.
-// (2) Port mapping definied by customers in the task definition.
+// (1) Ingress listeners for the service connect AppNet container in the service connect bridge network mode task.
+// (2) Port mapping configured by customers in the task definition.
+//
+// For service connect bridge mode task, we will create port bindings for customers' application containers
+// and service connect AppNet container, and let them to be published by the associated pause containers.
+// (a) For default bridge service connect experience, ECS Agent will assign a host port within the
+// default/user-specified dynamic host port range for the ingress listener. If no available host port can be
+// found by ECS Agent, an error will be returned.
+// (b) For non-default bridge service connect experience, ECS Agent will use the user-defined host port for the ingress listener.
 //
 // For non-service connect bridge network mode task, ECS Agent will assign a host port or a host port range
 // within the default/user-specified dynamic host port range. If no available host port or host port range can be
 // found by ECS Agent, an error will be returned.
 //
-// Note that,
+// Note that
 // (a) ECS Agent will not assign a new host port within the dynamic host port range for awsvpc network mode task
 // (b) ECS Agent will not assign a new host port within the dynamic host port range if the user-specified host port exists
 func (task *Task) dockerPortMap(container *apicontainer.Container, dynamicHostPortRange string) (nat.PortMap, error) {
 	hostPortStr := ""
 	dockerPortMap := nat.PortMap{}
-	scContainer := task.GetServiceConnectContainer()
 	containerToCheck := container
 	containerPortSet := make(map[int]struct{})
 	containerPortRangeMap := make(map[string]string)
+
+	// For service connect bridge network mode task, we will create port bindings for task containers,
+	// including both application containers and service connect AppNet container, and let them to be published
+	// by the associated pause containers.
 	if task.IsServiceConnectEnabled() && task.IsNetworkModeBridge() {
 		if container.Type == apicontainer.ContainerCNIPause {
-			// we will create bindings for task containers (including both customer containers and SC Appnet container)
-			// and let them be published by the associated pause container.
-			// Note - for SC bridge mode we do not allow customer to specify a host port for their containers. Additionally,
-			// When an ephemeral host port is assigned, Appnet will NOT proxy traffic to that port
+			// Find the task container associated with this particular pause container
 			taskContainer, err := task.getBridgeModeTaskContainerForPauseContainer(container)
 			if err != nil {
 				return nil, err
```

Comment:
nit(non-blocking): can also maybe add a logging statement here

Reply:
Thanks for catching this. Will fix the typo when we are going in the PR to rebase against the dev branch.

Reply:
Thanks for the suggestion! I saw that `getBridgeModeTaskContainerForPauseContainer()` has been used in `dockerExposedPorts()` and `PopulateServiceConnectContainerMappingEnvVar()` as well. These functions are related to [createContainer](https://github.com/aws/amazon-ecs-agent/blob/4943e4d853a84cf4d76b9d1241ee1f4f189b9c4a/agent/engine/docker_task_engine.go#L1288), thus error returned from `getBridgeModeTaskContainerForPauseContainer()` will eventually be logged as `dockerapi.DockerContainerMetadata{Error: XXX}` in ecs-agent.log.